### PR TITLE
Preliminary improvements to documentation, esp internal types e.g. Units

### PR DESCRIPTION
**Pull request by smallsaucepan**
Some preliminary work to improve documentation, particularly around internal types. Summary of changes:

#### Publish documentation for internal types

Internal types like Units, DbscanProps have had some JSDoc fields added. This will make them exportable as first class pages on the website (with some corresponding changes to turf-www) which we can link to from other pages that reference them.

They are also included in the README.md for the package in which they're defined. Not able to link directly between README.md files yet though.

#### Move away from categorising functions via documenation.yml

Suggest instead we add a custom JSDoc tag to relevant functions (@turfcategory), and use this to sort each function into its relevant category on the website e.g. along -> Measurement, hexGrid -> Grids. Category is co-located with the function. Less error prone as any function not listed in documents.yml ends up in Others, rather than being omitted entirely.

#### Sort categories alphabetically

This is largely to keep the web site generation code slightly simpler. With the sheer number of categories, plus better search, the order of the categories isn't that important.

#### Updated some mismatched parameter types

JSDoc parameter types that were out of sync with what the function actually used were updated where spotted. Particularly focusing on internal types whose documentation will be improving e.g. string -> Units.

#### CODE CHANGES

Only code change was to MoranIndex type, which is now being exported. To date this wasn't a named type and is an optional import so shouldn't cause any backwards compatibility issues.

### CODE REVIEW

**Review by twelch — COMMENTED** [review]
I like where this is going, nice work

File: packages/turf-along/index.ts [hunk 1]
```diff
@@ -8,11 +8,12 @@ import { getGeom } from "@turf/invariant";
 /**
  * Takes a {@link LineString} and returns a {@link Point} at a specified distance along the line.
  *
- * @name along
- * @param {Feature<LineString>} line input line
+ * @turfcategory Measurement
+ * @function
+ * @param {Feature<LineString>|LineString} line input line
```
[the remark this answers is not in the document]
Comment (smallsaucepan) [reply packages/turf-along/index.ts:13]: That's ok. This was one of the packages where the JSDoc was out of sync with the method signature - line can be a Feature or a bare Geometry.

File: packages/turf-sample/README.md [hunk 2]
```diff
@@ -8,8 +8,9 @@ Takes a [FeatureCollection][1] and returns a FeatureCollection with given number
 
 ### Parameters
 
-*   `featurecollection` **[FeatureCollection][1]** set of input features
+*   `fc` **[FeatureCollection][1]\<T>**&#x20;
```
[the remark this answers is not in the document]
Comment (smallsaucepan) [reply packages/turf-sample/README.md:11]: Yeah, haven't been able to track that down yet. It's unicode whitespace but not sure if documentation.js adding it, or if it's some junk in the source file. Will try to eliminate in a coming commit.

**Comment by smallsaucepan** [comment]
Thanks for taking a look @twelch. Sadly doesn't look like I'll be able to do the @turfcategory thing as documentation lint checks for unrecognised tags. Will stick with documentation.yml instead though make its usage more robust.

File: packages/turf-sample/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (smallsaucepan) [reply packages/turf-sample/README.md:11]: These look like they appeared around the time we upgraded to documentation.js 14.0.3. They're whitespace elements that are inserted when a field is empty, most notably when a comment isn't provided.

Harmless from what I can tell. Happy to leave them as is?

[the remark this answers is not in the document]
Comment (twelch) [reply packages/turf-sample/README.md:11]: Okay with me if left as-is

**Comment by smallsaucepan** [comment]
All I need to do now is figure out how I've broken the build by updating documentation 🙌

**Comment by smallsaucepan** [comment]
Got the build fixed @twelch if you're able to take another look.

**Review by twelch — APPROVED** [review]

**Pull request merged**